src/MoBi.Core/Domain/UnitSystem/DimensionFactoryExtensions.cs

```diff
@@ -13,6 +13,7 @@ public static IDimension TryGetDimension(this IDimensionFactory dimensionFactory
          }
          catch (KeyNotFoundException)
          {
+            // We could make the fix once here rather than per-usage
```

Comment:
This extension method is only used in these three places where I had to explicitly check for null. Maybe I can just move the `NoDimension` here instead of doing this 3 separate times?

Reply:
Depends if there are any other usages of the function where `null `is expected in case of non-existing dimension

Reply:
There are no other usages, only the 3 I modified to handle null.